packages/widgetbook_cli/bin/commands/publish.dart

```diff
@@ -0,0 +1,313 @@
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import 'dart:io';
+
+import 'package:ci/ci.dart' as ci;
+import 'package:mason_logger/mason_logger.dart';
+import 'package:path/path.dart' as p;
+import 'package:widgetbook_git/widgetbook_git.dart';
+
+import '../command.dart';
+import '../git-provider/github/github.dart';
+import '../models/deployment_data.dart';
+import '../parsers/exceptions.dart';
+import '../review/devices/device_parser.dart';
+import '../review/locales/locales_parser.dart';
+import '../review/text_scale_factors/text_scale_factor_parser.dart';
+import '../review/themes/theme_parser.dart';
+import '../review/use_cases/models/changed_use_case.dart';
+import '../review/use_cases/use_case_parser.dart';
+import '../widgetbook_http_client.dart';
+import '../widgetbook_zip_encoder.dart';
+
+class PublishCommand extends WidgetbookCommand {
+  PublishCommand({
+    super.logger,
+  }) {
+    argParser
+      ..addOption(
+        'path',
+        help: 'The path to the build folder of your application.',
+        defaultsTo: './',
+      )
+      ..addOption(
+        'api-key',
+        help: 'The project specific API key for Widgetbook Cloud.',
+        mandatory: true,
+      )
+      ..addOption(
+        'branch',
+        help: 'The name of the branch for which the Widgetbook is uploaded.',
+      )
+      ..addOption(
+        'repository',
+        help:
+            'The name of the repository for which the Widgetbook is uploaded.',
+      )
+      ..addOption(
+        'commit',
+        help:
+            'The SHA hash of the commit for which the Widgetbook is uploaded.',
+      )
+      ..addOption(
+        'actor',
+        help: 'The username of the actor which triggered the build.',
+      )
+      ..addOption(
+        'git-provider',
+        help: 'The name of the Git provider.',
+        defaultsTo: 'Local',
+        allowed: [
+          'GitHub',
+          'GitLab',
+          'BitBucket',
+          'Azure',
+          'Local',
+        ],
+      )
+      ..addOption(
+        'base-branch',
+        help:
+            'The base branch of the pull-request. For example, main or master.',
+      )
+      ..addOption(
+        'base-commit',
+        help: 'The SHA hash of the commit of the base branch.',
+      )
+      ..addOption(
+        'github-token',
+        help: 'GitHub API token.',
+      )
+      ..addOption(
+        'pr',
+        help: 'The number of the PR.',
+      );
+  }
+
+  @override
+  final String description = 'Publish a new build';
+
+  @override
+  final String name = 'publish';
+
+  @override
+  Future<int> run() async {
+    final publishProgress = logger.progress(
+      'Uploading build',
+    );
+    void publishProgressFailure() => publishProgress.fail();
+    final path = results['path'] as String;
+
+    if (!await GitDir.isGitDir(path)) {
+      publishProgressFailure();
+      usageException('Directory from "path" is not a Git folder');
+    }
```

Comment:
This error handling looks weird on the first look

Reply:
Ok, it's fine

Reply:
Hi, aha the  `!await GitDir.isGitDir(path)` returns a bool
. If the path is not a git folder (false) `usageException()` throws a Usage Exception. it is an internal handling error from `args`.